x-pack/plugins/fleet/public/components/agent_enrollment_flyout/agent_enrollment_flyout.test.tsx

```diff
@@ -29,7 +29,7 @@ import {
   KibanaVersionContext,
 } from '../../hooks';
 
-import { useFleetServerInstructions } from '../../applications/fleet/sections/agents/agent_requirements_page/components';
+import { useAdvancedForm } from '../../applications/fleet/components/fleet_server_instructions/hooks';
```

Comment:
I'm working on figuring out failing tests in this file and not having much luck. @hop-dev and I spent some time offline trying to figure out issues with mocking here that I'll try to summarize below. 

When the `ManagedInstructions` component is eventually rendered, it calls the `AgentPolicySelectionStep` function to generate a props object for `EuiSteps`. This `AgentPolicySelectionStep` is mocked in `agent_enrollment_flyout.test.mocks.ts` above. 

It seems like, based on our primitive debugging efforts, the instance of the mocked function that's being asserted on in these tests, and the instance that's being called in the actual `ManagedInstructions` component are different, so the assertion that the mock is called fails, despite us seeing multiple calls to the mock in the actual component file. 

We spent some time toying with various different approaches to mocking without much luck. My other thought is that (because we have multiple renders of `ManagedInstructions` here) there's some effect or state transition firing that's causing the assertion to happen before the component is actually done rendering. 

Quite a confusing issue, considering how little we actually touched these tests in this PR. Would love any input anyone might have 🙂 

@criamico - I have a feeling you are probably touching these tests in your work on the agent flyout as well. I'd be curious to see if you run into any similar issues here.

Reply:
I've made some progress here and gotten all but one test passing by asserting on the actual content of the steps rather than assert the functions that generate those steps are called. 

However, the one failing test that remains seems to be suffering from an odd timing issue. Will continue to investigate and update here.

Reply:
I did manage to eventually get all the tests passing here. Resolving.